src/pci_lmt/results.py

```diff
@@ -104,30 +104,32 @@ def write(self, result: LmtLaneResult) -> None:
         print(row)
 
 
-class OctTvOutputReporter(Reporter):
+class OcptvReporter(Reporter):
     def __init__(self):
-        # TODO(sksekar): Add support for saving to file.
-        self._run: tv.TestRun = None
-        self._step: tv.TestStep = None
+        self._run = tv.TestRun(name="pci_lmt", version=PCI_LMT_VERSION)
+        self._step = None
 
-    def start_run(self, host: HostInfo, version: str) -> None:
+    @contextmanager
+    def start_run(self, host: HostInfo):
         # TODO(sksekar): Add support for HardwareInfo using PlatformConfig.
-        self._run = tv.TestRun(name="pci_lmt", version=version)
         dut = tv.Dut(id=host.host_id, name=host.hostname)
-        self._run.start(dut=dut)
-
-    def end_run(self) -> None:
-        # TODO(sksekar): Add support for checking actual result based on error count.
-        self._run.end(status=TestStatus.COMPLETE, result=TestResult.PASS)
-
-    def start_step(self, name: str) -> None:
+        try:
+            yield self._run.start(dut=dut)
+        except TestRunError as e:
+            self._run.end(status=e.status, result=e.result)
+        else:
+            self._run.end(status=TestStatus.COMPLETE, result=TestResult.PASS)
+
+    @contextmanager
+    def start_step(self, name: str):
         # TODO(sksekar): Add support for validators.
         self._step = self._run.add_step(name=name)
```

Comment:
you dont need this `self._step` anymore; removing it will avoid the initial "lost step" problem.

this func can simply be
```
@contextmanager
def start_step(self, name: str):
  step = self._run.add_step(name)
  with step.scope():
    yield
```

Reply:
similar thing in `start_run`, so you dont have to deal with the exceptions

Reply:
> you dont need this self._step anymore; removing it will avoid the initial "lost step" problem.

I need to have self._step (as an object variable), since it's being used in the "write" method for adding measurement ("self._step.add_measurement"). One way to avoid is to perform "add_step" outside of the wrapper class (like it's done in the measurements.py you pointed earlier). Let me make the change.

> similar thing in start_run, so you dont have to deal with the exceptions

I'm not sure how I can avoid self._run unless I create the TestRun object outside of the wrapper class too. I prefer to have this as-is. Any concerns?

Reply:
yeah, had a feeling i was missing something

> I need to have self._step (as an object variable), since it's being used in the "write"

since you need to satisfy this `Reporter` interface, then id suggest a hard fail when trying to create a new step when the last one wasnt closed. Your api as is, with the yield, does not stop you from doing something like

```
r = Reporter()
with r.start_step("step1"):
  with r.start_step("step2"):
    pass
  r.write()  # this will incorrectly use step2
```
hence why `start_step("step2")` should fail with an exception here, at least to let the user know theyre using it incorrectly. There are ways of structurally doing this, but maybe that's too complicated at the moment.

> TestRun object outside of the wrapper class too. I prefer to have this as-is. Any concerns?

you can use the "throw invalid usage" exception pattern here too

Reply:
> then id suggest a hard fail when trying to create a new step when the last one wasnt closed.

Thanks for pointing this out with a valid test-case. Handled this for TestStep by adding a check for None.

I didn't change anything for TestRun since I don't think it's needed. I believe the TestRun is per test so the chances of user having multiple instances are low. Thoughts?

Reply:
yeah, i suppose this being python, having these runtime checks should be sufficient to remove the risk of having to maintain this code unnecessarily later on.

last version is ok, if you want to merge, i'll have to look at the ocptv api types failure

Reply:
Thanks for the feedback. If you are OK, i would like to merge the latest changes so that I can continue to work on validators and integrate more OCP-TV features.

Reply:
give me a day or 2 to fix the api types so we have all the tests working here